charts/k8s-service/templates/_statefulset_spec.tpl

```diff
@@ -0,0 +1,417 @@
+        - name: {{ $key }}
+{{ toYaml $value | indent 10 }}
+        {{- end }}
+
+
+    {{- if gt (len .Values.initContainers) 0 }}
+      initContainers:
+        {{- range $key, $value := .Values.initContainers }}
+        - name: {{ $key }}
+{{ toYaml $value | indent 10 }}
+        {{- end }}
+    {{- end }}
+
+    {{- /* START IMAGE PULL SECRETS LOGIC */ -}}
+    {{- if gt (len .Values.imagePullSecrets) 0 }}
+      imagePullSecrets:
+        {{- range $secretName := .Values.imagePullSecrets }}
+        - name: {{ $secretName }}
+        {{- end }}
+    {{- end }}
+    {{- /* END IMAGE PULL SECRETS LOGIC */ -}}
+
+    {{- /* START TERMINATION GRACE PERIOD LOGIC */ -}}
+    {{- if .Values.terminationGracePeriodSeconds }}
+      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
+    {{- end}}
+    {{- /* END TERMINATION GRACE PERIOD LOGIC */ -}}
+
+    {{- /* START VOLUME LOGIC */ -}}
+    {{- if index $hasInjectionTypes "hasVolume" }}
+      volumes:
+    {{- end }}
+    {{- range $name, $value := .Values.configMaps }}
+      {{- if eq $value.as "volume" }}
+        - name: {{ $name }}-volume
+          configMap:
+            name: {{ $name }}
+            {{- if $value.items }}
+            items:
+              {{- range $configKey, $keyMountConfig := $value.items }}
+              - key: {{ $configKey }}
+                path: {{ required "filePath is required for configMap items" $keyMountConfig.filePath | quote }}
+                {{- if $keyMountConfig.fileMode }}
+                mode: {{ include "k8s-service.fileModeOctalToDecimal" $keyMountConfig.fileMode }}
+                {{- end }}
+              {{- end }}
+            {{- end }}
+      {{- end }}
+    {{- end }}
+    {{- range $name, $value := .Values.secrets }}
+      {{- if eq $value.as "volume" }}
+        - name: {{ $name }}-volume
+          secret:
+            secretName: {{ $name }}
+            {{- if $value.items }}
+            items:
+              {{- range $secretKey, $keyMountConfig := $value.items }}
+              - key: {{ $secretKey }}
+                path: {{ required "filePath is required for secrets items" $keyMountConfig.filePath | quote }}
+                {{- if $keyMountConfig.fileMode }}
+                mode: {{ include "k8s-service.fileModeOctalToDecimal" $keyMountConfig.fileMode }}
+                {{- end }}
+              {{- end }}
+            {{- end }}
+      {{- end }}
+    {{- end }}
+    {{- range $name, $value := .Values.persistentVolumes }}
+        - name: {{ $name }}
+          persistentVolumeClaim:
+            claimName: {{ $value.claimName }}
+    {{- end }}
+    {{- range $name, $value := .Values.scratchPaths }}
+        - name: {{ $name }}
+          emptyDir:
+            medium: "Memory"
+    {{- end }}
+    {{- range $name, $value := .Values.emptyDirs }}
+        - name: {{ $name }}
+          emptyDir: {}
+    {{- end }}
+    {{- /* END VOLUME LOGIC */ -}}
```

Comment:
**issue (blocking):** One key feature of stateful sets is the `volumeClaimTemplates.` What is the ability to configure this? Is it expected to happen via another section?

Reply:
@mateimicu Is this something that could be added in a future iteration of this feature? Maybe this PR represents the initial bare-bones support that @mloskot and @rvc-mf are already using in their deployments 🤔